File: TypeScript/Problem3.ts
function lengthOfLongestSubstring(s: string): number {
    let res = 0,i = 0, j = 0;
    const found = new Set();
    while(j<s.length){
        if(!found.has(s[j])){
            found.add(s[j++]);
            res = Math.max(res, j-i);
        }else{
            found.delete(s[i++]);
        }
    }
    return res;
};

File: Java/Problem3.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 0, i = 0, j = 0;
        Set<Character> lett = new HashSet<>();
        
        while (j < s.length()) {
            if (!lett.contains(s.charAt(j))) {
                lett.add(s.charAt(j));
                res = Math.max(res, j - i + 1);
                j++;
            } else {
                lett.remove(s.charAt(i));
                i++;
            }
        }
        return res;
    }
}